src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/MultipleField.jsx

```diff
@@ -22,48 +22,41 @@ const styles = theme => {
     };
 };
 
-const _mapDataForOnChange = dataToMap => {
-    return dataToMap.filter(item => item.data !== undefined).map(item => item.data);
-};
-
-const reducer = (state, action) => {
-    switch (action.type) {
-        case 'set':
-            return action.data;
-        case 'unMount':
-            return action.onChange(_mapDataForOnChange(state), undefined);
-        default:
-            throw new Error();
-    }
-};
-
-export const MultipleFieldCmp = ({classes, inputContext, field, onChange, formik: {values}}) => {
+export const MultipleFieldCmp = ({classes, editorContext, inputContext, field, onChange, formik: {values}}) => {
     const {t} = useTranslation();
     const [isInit, setInit] = useState(false);
+    const currentValue = useRef(undefined);
```

Comment:
I cannot understand why using this manipulation? I think we added complexity to code and it will cause some rendering and performance issues.

Reply:
We are using a ref to be able to read the value from closured function call.